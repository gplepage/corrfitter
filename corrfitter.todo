- Add a median() function (like in lsqfit.Accumulator) to Dataset.

- Fix assemble(): doesn't work now if template contains reg-expression symbols (like +) since the template elements are used as regular expressions in one place (grep()) but later as literal strings. Must be all the same.