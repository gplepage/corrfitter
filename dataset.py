#!/usr/bin/env python
# encoding: utf-8
"""
dataset.py --- simplified replacement for old module; for legacy purposes only 
               (use gvar.Dataset for new stuff)

Created by Peter Lepage on 2012-05-22.
Copyright (c) 2012 Cornell University. All rights reserved.
"""

import gvar

class Dataset(gvar.Dataset):
    """ legacy class --- don't use otherwise """
    def __init__(self, *args,**kargs):
        super(Dataset, self).__init__(*args,**kargs)
    ##
    def gdev(self):
        return gvar.avg_data(self)
    ##
    def bin(self,nbin=2):
        return gvar.bin_data(self,binsize=nbin)
    ##
##  

if __name__ == '__main__':
    a = Dataset()
    a.append(x=1,y=[10,100])
    a.append(x=2,y=[20,200])
    a.append(x=3,y=[30,300])
    print (a.bin(3))

