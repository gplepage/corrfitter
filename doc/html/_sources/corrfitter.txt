:mod:`corrfitter` - Least-Squares Fit to Correlators
=====================================================

.. moduleauthor:: G.P. Lepage <g.p.lepage@cornell.edu>

.. |CorrFitter| replace:: :class:`corrfitter.CorrFitter`
.. |Corr2| replace:: :class:`corrfitter.Corr2`
.. |Corr3| replace:: :class:`corrfitter.Corr3`
.. |BaseModel| replace:: :class:`corrfitter.BaseModel`
.. |Dataset| replace:: :class:`gvar.dataset.Dataset`
.. |GVar| replace:: :class:`gvar.GVar`

.. automodule:: corrfitter
   :synopsis: Least-squares fitter for correlators.


Correlator Model Objects
-------------------------
Correlator objects describe theoretical models that are fit to
correlator data by varying the models' parameters. 

A model object's parameters are specified through priors for the fit. A
model assigns labels to each of its parameters (or arrays of related
parameters), and these labels are used to identify the corresponding
parameters in the prior. Parameters can be shared by more than one model
object.

A model object also specifies the data that it is to model. The data is 
identified by the data tag that labels it in the input file or |Dataset|.

.. autoclass:: corrfitter.Corr2
   :members:
   
.. autoclass:: corrfitter.Corr3
   :members:
   
.. autoclass:: corrfitter.BaseModel
   :members:


|CorrFitter| Objects
---------------------
|CorrFitter| objects are wrappers for :func:`lsqfit.nonlinear_fit()` which
is used to fit a collection of models to a collection of Monte Carlo data.

.. autoclass:: corrfitter.CorrFitter
   :members:

