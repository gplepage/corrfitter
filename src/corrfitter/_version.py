__version__ = '8.2'