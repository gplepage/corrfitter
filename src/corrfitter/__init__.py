"""
This module contains tools that facilitate least-squares fits, as functions
of time ``t``, of simulation (or other statistical) data for 2-point and
3-point correlators of the form::

    Gab(t)    =  <b(t) a(0)>
    Gavb(t,T) =  <b(T) V(t) a(0)>

where ``T > t > 0``. Each correlator is modeled using |Corr2| for 2-point
correlators, or |Corr3| for 3-point correlators in terms of amplitudes for
each source ``a``, sink ``b``, and vertex ``V``, and the energies
associated with each intermediate state. The amplitudes and energies are
adjusted in the least-squares fit to reproduce the data; they are defined
in a shared prior (typically a dictionary).

An object of type |CorrFitter| describes a collection of correlators and is
used to fit multiple models to data simultaneously. Fitting multiple
correlators simultaneously is important if there are statistical
correlations between the correlators. Any number of correlators may be
described and fit by a single |CorrFitter| object.

Typical code has the structure ::

    from corrfitter import CorrFitter

    data = make_data('mcfile')          # user-supplied routine
    models = make_models()              # user-supplied routine
    N = 4                               # number of terms in fit functions
    prior = make_prior(N)               # user-supplied routine
    fitter = CorrFitter(models=models)
    fit = fitter.lsqfit(data=data, prior=prior)  # do the fit
    print_results(fit, prior, data)     # user-supplied routine

where ``make_data`` assembles the correlator data to be fit,
``make_prior`` defines Bayesian priors for the fit parameters,
``fitter.lsqfit(...)`` does the fit, and ``print_results``
writes out the rsults. Sample code for each of these routines is
given in the tutorial documentation for |CorrFitter|.
"""

# Created by G. Peter Lepage, Cornell University, on 2010-11-26.
# Copyright (c) 2010-2021 G. Peter Lepage.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version (see <http://www.gnu.org/licenses/>).
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.

from ._corrfitter import * 
from ._version import __version__