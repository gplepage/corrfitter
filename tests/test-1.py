#!/usr/bin/env python
# encoding: utf-8
"""
test-1.py  --- standard usage (including error budgets and plots)

Created by Peter Lepage on 2010-11-26.
Copyright (c) 2010/2011 Cornell University. All rights reserved.
"""

import os
from corrfitter import Corr2,Corr3,CorrFitter
from gvar import gvar,log,exp,BufferDict
from gvar.dataset import Dataset,avg_data
from numpy import array,arange
import lsqfit
##
lsqfit.nonlinear_fit.fmt_parameter = '%7.3f +- %7.3f'

DISPLAYPLOTS = True         # display plots at end of fitting
try: 
    import matplotlib
except ImportError:
    DISPLAYPLOTS = False

TEST = True         # testing mode?

def main():
    dfile = "coarse_3pt_etas_Ds.bint8"   # data file
    data = avg_data(Dataset(dfile)) # compute avg and cov for data
    pfile = "test-1.p"                   # last fit stored in file test-1.p
    fitter = CorrFitter(models=build_models())
    nexp_list = NEXP_TEST if TEST else [2,3,4,5,6]
    p0 = P0_TEST if TEST else pfile
    for nexp in nexp_list:
        print '========================== nexp =',nexp
        prior = build_prior(nexp)
        fit = fitter.lsqfit(data=data,prior=prior,p0=p0) # pfile)
        print_results(fit,prior,data)
        print '\n\n'
    if DISPLAYPLOTS:
        fitter.display_plots()
##
      
def print_results(fit,prior,data):
    """ print out additional results from the fit """
    ## etas parameters ##
    dEetas = exp(fit.p['log(etas:dE)'])
    print 'etas:dE =',fmtlist(dEetas[:3])
    print ' etas:E =',fmtlist([sum(dEetas[:i+1]) for i in range(3)])
    aetas = exp(fit.p['log(etas:a)'])
    print ' etas:a =',fmtlist(aetas[:3])
    print
    ##
    ## Ds parameters ##
    dEDs = exp(fit.p['log(Ds:dE)'])
    print 'Ds:dE =',fmtlist(dEDs[:3])
    print ' Ds:E =',fmtlist([sum(dEDs[:i+1]) for i in range(3)])
    aDs = exp(fit.p['log(Ds:a)'])
    print ' Ds:a =',fmtlist(aDs[:3])
    print
    ##
    ## vertices ##
    print 'etas->V->Ds  =',fit.p['Vnn'][0,0].fmt()
    print 'etas->V->Dso =',fit.p['Vno'][0,0].fmt()
    print
    ##
    ## error budget ##
    outputs = dict(Vnn=fit.p['Vnn'][0,0],Vno=fit.p['Vno'][0,0],
                 Eetas=dEetas[0],EDs=dEDs[0])
    inputs = {'stat.':[data[k] for k in data]} # statistical errors in data
    inputs.update(prior)                       # all entries in prior
    print fit.fmt_partialsdev(outputs,inputs,ndigit=3)
    ##
##

def fmtlist(x):
    """ Make neat lists for printing. """
    return '  '.join([xi.fmt() for xi in x])
##
    
def build_prior(nexp):
    """ build prior """
    prior = BufferDict()
    ## etas ##
    prior.add('log(etas:a)',[log(gvar(0.3,0.3)) for i in range(nexp)])
    prior.add('log(etas:dE)',[log(gvar(0.5,0.5)) for i in range(nexp)])
    prior['log(etas:dE)'][0] = log(gvar(0.4,0.2))
    ##
    ## Ds ##
    prior.add('log(Ds:a)',[log(gvar(0.3,0.3)) for i in range(nexp)])
    prior.add('log(Ds:dE)',[log(gvar(0.5,0.5)) for i in range(nexp)])
    prior['log(Ds:dE)'][0] = log(gvar(1.2,0.2))
    prior.add('log(Ds:ao)',[log(gvar(0.1,0.1)) for i in range(nexp)])
    prior.add('log(Ds:dEo)',[log(gvar(0.5,0.5)) for i in range(nexp)])
    prior['log(Ds:dEo)'][0] = log(exp(prior['log(Ds:dE)'][0])+gvar(0.3,0.3))
    ##
    ## V ##
    prior.add('Vnn',[[gvar(0.01,1.0) for i in range(nexp)] for j in range(nexp)])
    prior.add('Vno',[[gvar(0.01,1.0) for i in range(nexp)] for j in range(nexp)])
    return prior
##

def build_models():
    """ build models """
    tmin = 5
    tdata = range(64)
    tfit = range(tmin,64-tmin) # all ts
    
    tp = 64 # periodic
    models = [
        Corr2(datatag='etas',tp=tp,tdata=tdata,tfit=tfit,
            a='log(etas:a)',b='log(etas:a)',dE='log(etas:dE)'),
            
        Corr2(datatag='Ds',tp=tp,tdata=tdata,tfit=tfit,
            a=('log(Ds:a)','log(Ds:ao)'),b=('log(Ds:a)','log(Ds:ao)'),
            dE=('log(Ds:dE)','log(Ds:dEo)'),s=(1.,-1.)),

        Corr3(datatag='3ptT15',tdata=range(16),T=15,tfit=range(tmin,16-tmin),
            a='log(etas:a)',dEa='log(etas:dE)',tpa=tp,
            b=('log(Ds:a)','log(Ds:ao)'),dEb=('log(Ds:dE)','log(Ds:dEo)'),tpb=tp,sb=(1,-1.),
            Vnn='Vnn',Vno='Vno',transpose_V=False),
            
        Corr3(datatag='3ptT16',tdata=range(17),T=16,tfit=range(tmin,17-tmin),
            a='log(etas:a)',dEa='log(etas:dE)',tpa=tp,
            b=('log(Ds:a)','log(Ds:ao)'),dEb=('log(Ds:dE)','log(Ds:dEo)'),tpb=tp,sb=(1,-1.),
            Vnn='Vnn',Vno='Vno',transpose_V=False)
    ]
    return models
##    

NEXP_TEST = [6]
P0_TEST = {'log(Ds:a)': [-1.5382029985053618, -1.2890389183673718,
-0.8266404705754684, -1.1281693592882478, -1.1942970642079684,
-1.2029575678081768], 'log(Ds:ao)': [-2.7187514959968135, -2.4930018835083185,
-2.153056999309124, -2.274349119294338, -2.3000607164691345,
-2.3023830129920464], 'Vno': [[-0.7767194750934922, 0.29893034835380455,
0.0018184442826528018, 0.021717846191188498, 0.011730183184535546,
0.010175999805737345], [0.21499388626034838, 0.08090442617659241,
0.004526958942468914, 0.009566576678925012, 0.009981460666758885,
0.009999494947514592], [-0.03929180298702926, 0.016001668510532977,
0.010207619966221364, 0.009993775842607086, 0.009999490734868723,
0.009999977541285848], [-0.0008212538513334825, 0.009446705426502066,
0.010011631248341204, 0.010000150342742585, 0.009999996724907924,
0.009999999715913033], [0.008707361530432445, 0.009892605564266226,
0.009999973201624102, 0.010000008998777822, 0.01000000009622645,
0.009999999998091305], [0.009873335420234476, 0.00998801754685697,
0.009999941109021849, 0.010000000137364686, 0.010000000005831669,
0.010000000000057843]], 'Vnn': [[0.7674463006011443, -0.5085825363348335,
0.45476872843648597, 0.07910330776006197, 0.01672199952261604,
0.010575060501324804], [0.0428274543801929, 0.2276246794008917,
0.024339418136236402, 0.011105918132437442, 0.010079750081074862,
0.01000540600309465], [-0.0789057702177111, 0.039088907022644885,
0.010402246249041023, 0.010015038267077255, 0.010001273028618115,
0.010000092479777158], [-0.04722804062576639, 0.01216744637810462,
0.01003297837759048, 0.010000246006116778, 0.010000007960775022,
0.010000000698523368], [0.00035115678498912886, 0.010139057728244761,
0.01000276274869412, 0.010000023410840267, 0.010000000149913561,
0.01000000000466716], [0.008870833254729392, 0.01000779450180237,
0.010000194918187728, 0.010000001937760121, 0.01000000001438937,
0.01000000000008824]], 'log(Ds:dEo)': [0.36852247027597856,
-1.450434416125243, -0.716397528343509, -0.7262479860745519,
-0.6965486249012215, -0.6934059166295101], 'log(etas:dE)':
[-0.8765386251062858, -0.44666827867143977, -0.7029624277436295,
-0.6648624116442549, -0.6824061676451147, -0.6916868235591809], 'log(Ds:dE)':
[0.18374400617661235, -0.6825021780114466, -0.669942670009535,
-0.7649157041649213, -0.7045972594319748, -0.6943318698851685], 'log(etas:a)':
[-1.5213670965733321, -1.5776623136419392, -1.2117105815007296,
-1.2436876034918443, -1.213887220332993, -1.205310268830514]}

# for k in P0_TEST:
#     P0_TEST[k] = array(P0_TEST[k])

if __name__ == '__main__':
    if True:
        main()
    else:
        import hotshot, hotshot.stats
        prof = hotshot.Profile("lsqfit-test1.prof")
        prof.runcall(main)
        prof.close()
        stats = hotshot.stats.load("lsqfit-test1.prof")
        stats.strip_dirs()
        stats.sort_stats('time', 'calls')
        stats.print_stats(140)

