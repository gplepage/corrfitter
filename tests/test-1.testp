(dp0
S'log(Ds:a)'
p1
cnumpy.core.multiarray
_reconstruct
p2
(cnumpy
ndarray
p3
(I0
tp4
S'b'
p5
tp6
Rp7
(I1
(I6
tp8
cnumpy
dtype
p9
(S'f8'
p10
I0
I1
tp11
Rp12
(I3
S'<'
p13
NNNI-1
I-1
I0
tp14
bI00
S'\xdf\x08S\xbfz\x9c\xf8\xbf\x93\x8b\xdaE\xe7\x9f\xf4\xbfv~U\xb7\xd6s\xea\xbft\xe0gP\xfb\x0c\xf2\xbfl\xb6\x07=\xd7\x1b\xf3\xbf\x9fcCoP?\xf3\xbf'
p15
tp16
bsS'log(Ds:ao)'
p17
g2
(g3
(I0
tp18
g5
tp19
Rp20
(I1
(I6
tp21
g12
I00
S'"\x0f\xca\xc8\x00\xc0\x05\xc0\xecF\xb4\xf8\xaa\xf1\x03\xc0\x15\xa7\xb3\xf2u9\x01\xc0\xd6qx\xf3\xdd1\x02\xc0\xe3d\xa0;\x86f\x02\xc0\x92U\xfd\xc8Gk\x02\xc0'
p22
tp23
bsS'Vno'
p24
g2
(g3
(I0
tp25
g5
tp26
Rp27
(I1
(I6
I6
tp28
g12
I00
S"\xb5+\xf6\xcc\xe2\xda\xe8\xbf\xdb\x88}\xc1\xac!\xd3?\xabTO\xae\x1b\xcb]?\x91\xd9\x85\xfc3=\x96?\xf9\xb6:\x89\xfe\x05\x88?y<\xf5\x92'\xd7\x84?~\x00*o\xeb\x84\xcb?V\xb5\x87\x08'\xb6\xb4?\x89X\xbbI\xdc\x8ar?\xfb\xe8\xc3/\xa4\x97\x83?JR\n\xf9(q\x84?\xf0o5~\x9dz\x84?}\x0b\xa9!\x0e\x1e\xa4\xbf\xa1\x97\xbc\xcb\xbdb\x90?\x16;f\x8f\xbb\xe7\x84?\xc5\xaaB\xe3\x9dw\x84?\xc1\x9cv\xed\x9cz\x84?#$\x01D\xdez\x84?t\xd0r7-\xe9J\xbf\xc6\xe8\xe4V\xcbX\x83?!tTf\xfa\x80\x84?9\xd4ju\xf5z\x84?\xc1\x10&\xd7\xe0z\x84?\x809\xeb=\xe1z\x84?\xe0\xad\x0fH*\xd5\x81?VL\xf7\n\x93B\x84?\xa0\x12\xe5\xae\xddz\x84?\xb4*\xe0|\xe2z\x84?/\x7f\xfcJ\xe1z\x84?{J\x9dG\xe1z\x84?\xf4\xd7\xdb\xa5x8\x84?\x83\x99\xb9\x05\x99t\x84?\xda\x8e3`\xd9z\x84?AZfL\xe1z\x84?;`\xe1G\xe1z\x84?\xbb\x96\xaeG\xe1z\x84?"
p29
tp30
bsS'Vnn'
p31
g2
(g3
(I0
tp32
g5
tp33
Rp34
(I1
(I6
I6
tp35
g12
I00
S'F\x94P\x8b\xeb\x8e\xe8?\xb4\xfe\x1b\xe2NF\xe0\xbf\xa24\xf8K\xee\x1a\xdd?L\xbb\xd5G\x1d@\xb4?\xcd\x1fKd\x92\x1f\x91?k!\r\x98`\xa8\x85?\xf5"\xde\xe7z\xed\xa5?D\x08\xe54\xce"\xcd?=`\x98\xb3n\xec\x98?@\xf2]\x19\xb3\xbe\x86?=\xbe\x99\'\xb1\xa4\x84?\xcb-\x88\xdc\xb6}\x84?\x9b\xfe\x8c&+3\xb4\xbf\xef\xe7\x98\x12v\x03\xa4?\xa5z\x8e\xdb\xc5M\x85?1/\x9a\xae\xc3\x82\x84?\x919\x9c$\x8c{\x84?t\xcdB\xb1\xedz\x84?\x057\xe2\x13F.\xa8\xbf;\x89(\x02?\xeb\x88?]\x16\x1b\x90+\x8c\x84?h\xc4bL\x02{\x84?1\xcc5Y\xe2z\x84?E[\xae_\xe1z\x84?\x98\x9aD\xe8n\x037?\x87-\xd7J\xc9\xc3\x84?)\x97\x00\x17T|\x84?;\x02\x12l\xe4z\x84?\xcf\xbb\xd4L\xe1z\x84?\xfd!\xd7G\xe1z\x84?\x8c\x18\xbd\x15\xdf*\x82?\xd3\xfe\xb8p\xf7~\x84?J\x97\x04q\xfbz\x84?\\\xccB\x8a\xe1z\x84?v\xa6,H\xe1z\x84?.\xdb\xaeG\xe1z\x84?'
p36
tp37
bsS'log(Ds:dEo)'
p38
g2
(g3
(I0
tp39
g5
tp40
Rp41
(I1
(I6
tp42
g12
I00
S'\x1fKkE\xdf\x95\xd7?+\x03\xe4\xb7\xfa4\xf7\xbf\x97ue\x82\xba\xec\xe6\xbf\xea;\x9fjl=\xe7\xbf\xd9\xc8\x80W J\xe6\xbfN\xda\xd8\x9aa0\xe6\xbf'
p43
tp44
bsS'log(etas:dE)'
p45
g2
(g3
(I0
tp46
g5
tp47
Rp48
(I1
(I6
tp49
g12
I00
S'\xc7d\x10\xbb\x9a\x0c\xec\xbf\x14|C\x8c6\x96\xdc\xbf\x8a8\xaf\x0f\xab~\xe6\xbf\xa7AK\x89\x8dF\xe5\xbf\x96\xfb\x93uE\xd6\xe5\xbf\x82X\xc8gL"\xe6\xbf'
p50
tp51
bsS'log(Ds:dE)'
p52
g2
(g3
(I0
tp53
g5
tp54
Rp55
(I1
(I6
tp56
g12
I00
S'L\xaa\xaep\xec\x84\xc7?\xfe@\xc0\xce\x0e\xd7\xe5\xbf\x12Y<\x9c+p\xe5\xbf\x88\xb9\xb2\x7f0z\xe8\xbf\xe5\x91C\x8d\x0f\x8c\xe6\xbfAF7x\xf77\xe6\xbf'
p57
tp58
bsS'log(etas:a)'
p59
g2
(g3
(I0
tp60
g5
tp61
Rp62
(I1
(I6
tp63
g12
I00
S'\x10\xe3O\x06\x85W\xf8\xbf\xc8\x94\x93\xd6\x1a>\xf9\xbfU4|\xa2*c\xf3\xbfg\x02\xf7\xf8$\xe6\xf3\xbf\x02\xcd\x85\x01\x15l\xf3\xbfw\x8f\xa2k\xf3H\xf3\xbf'
p64
tp65
bs.