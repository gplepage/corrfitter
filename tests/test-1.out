========================== nexp = 6
Least Square Fit:
  chi2/dof [dof] = 0.6 [69]    Q = 1    logGBF = -141.36    itns = 1

Parameters:
    log(etas:a)_ -1.521 +-   0.000           (    -1.2 +-        1)
               _ -1.578 +-   0.258           (    -1.2 +-        1)
               _ -1.212 +-   0.593           (    -1.2 +-        1)
               _ -1.244 +-   0.972           (    -1.2 +-        1)
               _ -1.214 +-   0.999           (    -1.2 +-        1)
               _ -1.205 +-   1.000           (    -1.2 +-        1)
   log(etas:dE)_ -0.877 +-   0.000           (   -0.92 +-      0.5)
               _ -0.447 +-   0.093           (   -0.69 +-        1)
               _ -0.703 +-   0.746           (   -0.69 +-        1)
               _ -0.665 +-   0.971           (   -0.69 +-        1)
               _ -0.682 +-   0.999           (   -0.69 +-        1)
               _ -0.692 +-   1.000           (   -0.69 +-        1)
      log(Ds:a)_ -1.538 +-   0.001           (    -1.2 +-        1)
               _ -1.289 +-   0.094           (    -1.2 +-        1)
               _ -0.827 +-   0.420           (    -1.2 +-        1)
               _ -1.128 +-   0.984           (    -1.2 +-        1)
               _ -1.194 +-   1.000           (    -1.2 +-        1)
               _ -1.203 +-   1.000           (    -1.2 +-        1)
     log(Ds:dE)_  0.184 +-   0.000           (    0.18 +-     0.17)
               _ -0.683 +-   0.041           (   -0.69 +-        1)
               _ -0.670 +-   0.404           (   -0.69 +-        1)
               _ -0.765 +-   0.986           (   -0.69 +-        1)
               _ -0.705 +-   1.000           (   -0.69 +-        1)
               _ -0.694 +-   1.000           (   -0.69 +-        1)
     log(Ds:ao)_ -2.719 +-   0.118           (    -2.3 +-        1)
               _ -2.493 +-   0.395           (    -2.3 +-        1)
               _ -2.153 +-   0.795           (    -2.3 +-        1)
               _ -2.274 +-   0.998           (    -2.3 +-        1)
               _ -2.300 +-   1.000           (    -2.3 +-        1)
               _ -2.302 +-   1.000           (    -2.3 +-        1)
    log(Ds:dEo)_  0.369 +-   0.010           (    0.41 +-     0.24)
               _ -1.450 +-   0.505           (   -0.69 +-        1)
               _ -0.716 +-   0.783           (   -0.69 +-        1)
               _ -0.726 +-   0.997           (   -0.69 +-        1)
               _ -0.697 +-   1.000           (   -0.69 +-        1)
               _ -0.693 +-   1.000           (   -0.69 +-        1)
            Vnn_  0.767 +-   0.001           (    0.01 +-        1)
               _ -0.509 +-   0.041           (    0.01 +-        1)
               _  0.455 +-   0.509           (    0.01 +-        1)
               _  0.079 +-   0.993           (    0.01 +-        1)
               _  0.017 +-   1.000           (    0.01 +-        1)
               _  0.011 +-   1.000           (    0.01 +-        1)
               _  0.043 +-   0.045           (    0.01 +-        1)
               _  0.228 +-   0.803           (    0.01 +-        1)
               _  0.024 +-   0.996           (    0.01 +-        1)
               _  0.011 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _ -0.079 +-   0.300           (    0.01 +-        1)
               _  0.039 +-   0.999           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _ -0.047 +-   0.988           (    0.01 +-        1)
               _  0.012 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.000 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.009 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
            Vno_ -0.777 +-   0.075           (    0.01 +-        1)
               _  0.299 +-   0.336           (    0.01 +-        1)
               _  0.002 +-   0.873           (    0.01 +-        1)
               _  0.022 +-   0.998           (    0.01 +-        1)
               _  0.012 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.215 +-   0.444           (    0.01 +-        1)
               _  0.081 +-   0.976           (    0.01 +-        1)
               _  0.005 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _ -0.039 +-   0.976           (    0.01 +-        1)
               _  0.016 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _ -0.001 +-   1.000           (    0.01 +-        1)
               _  0.009 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.009 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)

etas:dE = 0.41622(8)  0.64(6)  0.5(4)
 etas:E = 0.41622(8)  1.06(6)  1.6(4)
 etas:a = 0.2184(1)  0.21(5)  0.3(2)

Ds:dE = 1.2017(1)  0.51(2)  0.5(2)
 Ds:E = 1.2017(1)  1.71(2)  2.2(2)
 Ds:a = 0.2148(2)  0.28(3)  0.4(2)

etas->V->Ds  = 0.7674(9)
etas->V->Dso = -0.78(7)

Partial % Errors:
                           Vno       EDs     Eetas       Vnn
------------------------------------------------------------
          log(Ds:a):     0.475     0.001     0.000     0.017
        log(etas:a):     0.048     0.000     0.001     0.003
         log(Ds:ao):     3.854     0.001     0.000     0.006
                Vno:     2.712     0.002     0.000     0.010
                Vnn:     1.008     0.002     0.001     0.031
        log(Ds:dEo):     4.468     0.001     0.000     0.006
       log(etas:dE):     0.054     0.000     0.001     0.003
         log(Ds:dE):     0.270     0.001     0.000     0.021
              stat.:     7.028     0.011     0.019     0.109
------------------------------------------------------------
              total:     9.637     0.011     0.020     0.117




